src/analysis/special_functions/polynomials.lean

```diff
@@ -0,0 +1,137 @@
+  (λ x, eval x P) ~[at_top] (λ x, P.leading_coeff * x ^ P.nat_degree) :=
+begin
+  by_cases h : P = 0,
+  { simp [h] },
+  { conv_lhs
+    { funext,
+      rw [polynomial.eval_eq_finset_sum, sum_range_succ, add_comm] },
+    exact is_equivalent.refl.add_is_o (is_o.sum $ λ i hi, is_o.const_mul_left
+      (is_o.const_mul_right (λ hz, h $ leading_coeff_eq_zero.mp hz) $
+        is_o_pow_pow_at_top_of_lt (mem_range.mp hi)) _) }
+end
+
+lemma tendsto_at_top_of_leading_coeff_nonneg (P : polynomial α) (hdeg : 1 ≤ P.degree)
+  (hnng : 0 ≤ P.leading_coeff) : tendsto (λ x, eval x P) at_top at_top :=
+P.is_equivalent_at_top_lead.symm.tendsto_at_top
+  (tendsto_const_mul_pow_at_top (nat_degree_ge_of_degree_ge_coe hdeg)
+    (lt_of_le_of_ne hnng $ ne.symm $ mt leading_coeff_eq_zero.mp $ ne_zero_of_degree_ge_coe hdeg))
+
+lemma tendsto_at_bot_of_leading_coeff_nonpos (P : polynomial α) (hdeg : 1 ≤ P.degree)
+  (hnps : P.leading_coeff ≤ 0) : tendsto (λ x, eval x P) at_top at_bot :=
+P.is_equivalent_at_top_lead.symm.tendsto_at_bot
+  (tendsto_neg_const_mul_pow_at_top (nat_degree_ge_of_degree_ge_coe hdeg)
+    (lt_of_le_of_ne hnps $ mt leading_coeff_eq_zero.mp $ ne_zero_of_degree_ge_coe hdeg))
+
+lemma abs_tendsto_at_top (P : polynomial α) (hdeg : 1 ≤ P.degree) :
+  tendsto (λ x, abs $ eval x P) at_top at_top :=
+begin
+  by_cases hP : 0 ≤ P.leading_coeff,
+  { exact tendsto_abs_at_top_at_top.comp (P.tendsto_at_top_of_leading_coeff_nonneg hdeg hP)},
+  { push_neg at hP,
+    exact tendsto_abs_at_bot_at_top.comp (P.tendsto_at_bot_of_leading_coeff_nonpos hdeg hP.le)}
+end
+
+lemma is_equivalent_at_top_div {P Q : polynomial α} :
+   (λ x, (eval x P)/(eval x Q)) ~[at_top]
+     λ x, P.leading_coeff/Q.leading_coeff * x^(P.nat_degree - Q.nat_degree : ℤ) :=
+begin
+  by_cases hP : P = 0,
+  { simp [hP] },
+  by_cases hQ : Q = 0,
+  { simp [hQ] },
+  refine (P.is_equivalent_at_top_lead.symm.div
+          Q.is_equivalent_at_top_lead.symm).symm.trans
+         (eventually_eq.is_equivalent ((eventually_gt_at_top 0).mono $ λ x hx, _)),
+  simp [← div_mul_div, hP, hQ, fpow_sub hx.ne.symm]
+end
+
+lemma div_tendsto_zero_of_degree_lt (P Q : polynomial α) (hdeg : P.degree < Q.degree) :
+  tendsto (λ x, (eval x P)/(eval x Q)) at_top (𝓝 0) :=
+begin
+  by_cases hP : P = 0,
+  { simp [hP, tendsto_const_nhds] },
+  rw ←  nat_degree_lt_nat_degree_iff hP at hdeg,
+  refine is_equivalent_at_top_div.symm.tendsto_nhds _,
+  rw ← mul_zero,
+  refine tendsto.const_mul _ _,
+  apply tendsto_fpow_at_top_zero,
+  linarith
+end
+
+lemma div_tendsto_leading_coeff_div_of_degree_eq (P Q : polynomial α)
+  (hdeg : P.degree = Q.degree) :
+  tendsto (λ x, (eval x P)/(eval x Q)) at_top (𝓝 $ P.leading_coeff / Q.leading_coeff) :=
+begin
+  refine is_equivalent_at_top_div.symm.tendsto_nhds _,
+  rw show (P.nat_degree : ℤ) = Q.nat_degree, by simp [hdeg, nat_degree],
+  simp [tendsto_const_nhds]
+end
+
+lemma div_tendsto_at_top_of_degree_gt (P Q : polynomial α) (hdeg : Q.degree < P.degree)
```

Comment:
Can you also give a version where you replace the two assumptions with `0 < P.leading_coeff/Q.leading_coeff`, that might be quicker to check in practice? Same thing for the next lemma.